File: src/main/java/net/treset/minecraftlauncher/util/ui/sort/InstanceDetailsLastPlayedComparator.java
package net.treset.minecraftlauncher.util.ui.sort;

import net.treset.minecraftlauncher.LauncherApplication;
import net.treset.minecraftlauncher.data.InstanceData;

import java.util.Comparator;
import java.util.Objects;

public class InstanceDetailsLastPlayedComparator implements Comparator<InstanceData> {

    @Override
    public int compare(InstanceData o1, InstanceData o2) {
        if(Objects.equals(o1.getInstance().getValue().getLastPlayed(), o2.getInstance().getValue().getLastPlayed())) {
            return 0;
        }
        if(o1.getInstance().getValue().getLastPlayed() == null) {
            return 1;
        }
        if(o2.getInstance().getValue().getLastPlayed() == null) {
            return -1;
        }
        return o2.getInstance().getValue().getLastPlayed().compareTo(o1.getInstance().getValue().getLastPlayed());
    }

    @Override
    public String toString() {
        if(LauncherApplication.stringLocalizer == null) {
            return "sort.lastplayed";
        }
        return LauncherApplication.stringLocalizer.get("sort.lastplayed");
    }
}

File: client/composeApp/src/commonMain/kotlin/net/treset/treelauncher/backend/util/sort/InstanceDetailsLastPlayedComparator.kt
package net.treset.treelauncher.backend.util.sort

import net.treset.treelauncher.backend.data.InstanceData
import net.treset.treelauncher.localization.strings

class InstanceDetailsLastPlayedComparator : Comparator<InstanceData> {
    override fun compare(o1: InstanceData, o2: InstanceData): Int {
        val firstLastPlayed = o1.instance.lastUsed
        val secondLastPlayed = o2.instance.lastUsed

        if (firstLastPlayed == secondLastPlayed) {
            return 0
        }
        return secondLastPlayed.compareTo(firstLastPlayed)
    }

    override fun toString(): String = strings().sortBox.sort.lastPlayed()
}
